edb/server/protocol/binary.pyx

```diff
@@ -298,7 +298,10 @@ cdef class EdgeConnection:
             self._transport.abort()
             self._transport = None
 
-    async def close(self):
+    def close(self):
+        if self.dbview is not None:
```

Comment:
Hm, we need to put this in `connection_lost`.

Reply:
Done. I've added the necessary bits to `abort()` and made sure it always gets called.  The `pinned_pgcon` abort business has been moved there as well.